### PR TITLE
Add execution-id & separate the options BUILD targets

### PR DESCRIPTION
**Pull request by oschaaf**
Another prelude to distributed test execution. Being able to
identify executions and associate results is going to come in
handy.

Also, this separates some BUILD targets to allow for a leaner build
process of future things that are going to depend on it.

Signed-off-by: Otto van der Schaaf <ovanders@redhat.com>

### CODE REVIEW

**Comment by mum4k** [comment]
@wjuan-AFK please review and assign back to me once done.

**Review by wjuan-AFK — APPROVED** [review]

File: api/client/options.proto [hunk 1]
```diff
@@ -106,7 +106,7 @@ message H1ConnectionReuseStrategy {
 
 // TODO(oschaaf): Ultimately this will be a load test specification. The fact that it
 // can arrive via CLI is just a concrete detail. Change this to reflect that.
-// highest unused number is 39
+// highest unused number is 40
```
Comment (mum4k) [inline api/client/options.proto:109]: (pre-existing) Is this true? There seem to be id `105` just above it.

File: api/client/options.proto [hunk 2]
```diff
@@ -225,4 +225,10 @@ message CommandLineOptions {
   google.protobuf.Timestamp scheduled_start = 105;
 
   reserved 38; // deprecated
+  // Provide a unique execution id. Will be reflected in the output when set.
```
Comment (mum4k) [inline api/client/options.proto:228]: Is this a good opportunity to also describe the required format? E.g. the identifier must be something that can use used as a directory name on Linux filesystem?

**Review by mum4k — CHANGES_REQUESTED** [review]

File: api/client/options.proto (hunk 2 above)
Comment (oschaaf) [reply api/client/options.proto:228]: See below, `When populating this field, it is recommended...`. There will only be constraints to this value when there is a Sink involved which imposes limitations; our file based sink is picky (at least today) and only accepts unique identifiers. Maybe we can later work to remove that restriction by using a codec. For max key length I was thinking along the same lines, that might vary depending on the implementation specifics of Sinks?

File: api/client/options.proto (hunk 1 above)
Comment (oschaaf) [reply api/client/options.proto]: added cc80ec1e

File: api/client/options.proto (hunk 2 above)
Comment (mum4k) [reply api/client/options.proto:228]: Makes sense, we don't want to expose limitations of a single sink up to the public API.

We can make sure that the individual Sink implementations will validate their inputs and return readable error messages when an incompatible `execution_id` will be specified.

**Review by mum4k — APPROVED** [review]

**Pull request merged**